credentials/apps/badges/services/revocation.py

```diff
@@ -39,23 +36,18 @@ def process_penalties(event_type, username, payload_dict):
             - emit BADGE_PROGRESS_INCOMPLETE >> handle_badge_incompletion
     """
 
-    # TEMP: remove this stub after processing is implemented
-    if keypath(payload_dict, "course_passing_status.status") == CoursePassingStatusData.FAILING:
-        BADGE_PROGRESS_INCOMPLETE.send(
-            sender=None,
-            username=username,
-            badge_template_id=CredlyBadgeTemplate.objects.first().id,
-        )
-
-    # :TEMP
 
     penalties = discover_penalties(event_type=event_type)
-
-    # actual processing goes here:
-
-    # for penalty in penalties:
-    #     penalty.apply_rules(**kwargs)
-    #     penalty.assign(username)
+    for penalty in penalties:
+        if not penalty.is_active:
+            continue
+        if penalty.apply_rules(payload_dict):
+            [
```

Comment:
I'm looking for this:
```python
penalty.requirements.reset(username)

# or

for requirement in penalty.requirements.all():
    requirement.reset(username)
```

Reply:
e.g. a penalty can be set to affect not all badge progress (only some requirements)